pyzeebe/worker/worker.py

```diff
@@ -52,12 +53,21 @@ def work(self) -> None:
         for task in self.tasks:
             task_thread = Thread(target=self._handle_task, args=(task,))
             task_thread.start()
+            self._task_threads.append(task_thread)
 
-    def stop(self) -> None:
+    def stop(self, wait: bool = False) -> None:
         """
-        Stop the worker. This will wait for all tasks to complete before stopping
+        Stop the worker. This will emit a signal asking tasks to complete the current task and stop polling for new.
+
+        Args:
+            wait (bool): Wait for all tasks to complete
         """
         self.stop_event.set()
+        if wait:
+            logger.debug("Waiting for threads to join")
+            while self._task_threads:
```

Comment:
Is there a particular reason to use `while` instead of `for` here?

Reply:
The idea is that if any new items are added to the list, while processing the once that's there when it enters the loop, those will be joined as well.